### DISCUSSION TITLE
GEFS data source unavailable

### DISCUSSION THREAD

**Discussion by John-Polo** [microsoft/PlanetaryComputer#260] [Q&A] [answered]
I am trying to use GEFS data. https://planetarycomputer.microsoft.com/dataset/storage/noaa-gefs  

The documentation says that the data have to be accessed via an Azure blob. I'm learning the Azure processing for R with AzureStor library. 

At the moment, I'm just doing a basic import of the data and trying to parse the object to understand it better, but I can't get the data imported. My code so far is
`library(AzureStor)`
`endp <- storage_endpoint("https://noaagefs.blob.core.windows.net/gefs/gefs.20230822/06/atmos/pgrb2sp25/geavg.t06z.pgrb2s.0p25.f009")`

`cont <- blob_container(endp, "gefs")`

`list_blobs(cont)`

I get this error:

> Error in process_storage_response(response, match.arg(http_status_handler), : Bad Request (HTTP 400). Failed to complete Storage Services operation. Message:
InvalidUri
The requested URI does not represent any resource on the server.
RequestId:911629d9-e01e-0029-15f8-d5fe4a000000
Time:2023-08-23T19:35:30.3704819Z.

I pasted the URL in my browser and I download a file, so it seems to me that there is some kind of resource. What am I doing wrong in Planetary Computer?

**Comment by TomAugspurger** [comment, marked as answer]
Sorry, I've never used the `AzureStor` library before :/

Glancing at the readme, though, you might want something like

```R
storage_endpoint("https://noaagefs.blob.core.windows.net")
cont <- blob_container(endp, "gefs")
list_blobs(cont)
```

The storage account is `https://noaagefs.blob.core.windows.net`.  The URL you had in there is a full Blob URL.

**Reply by John-Polo** [reply]
The URL I supplied was the problem. Thank you, Tom.
